symengine/eval_double.cpp

```diff
@@ -243,6 +243,16 @@ class EvalRealDoubleVisitor : public BaseVisitor<EvalRealDoubleVisitor, EvalDoub
     };
 };
 
+class EvalRealDoubleVisitorFinal : public BaseVisitor<EvalRealDoubleVisitorFinal, EvalRealDoubleVisitor> {
+public:
+    using EvalRealDoubleVisitor::bvisit;
+
+    double apply(const Basic &b) {
+        b.accept_EvalRealDoubleVisitorFinal(*this);
```

Comment:
@certik, this is not going to work, as all the `bvisit` methods are going to be calling `EvalRealDoubleVisitor`'s apply method